validator_client/validator_services/src/attestation_service.rs

```diff
@@ -207,24 +247,43 @@ impl<S: ValidatorStore + 'static, T: SlotClock + 'static> AttestationService<S,
                 map
             });
 
-        // For each committee index for this slot:
-        //
-        // - Create and publish an `Attestation` for all required validators.
-        // - Create and publish `SignedAggregateAndProof` for all aggregating validators.
-        duties_by_committee_index
-            .into_iter()
-            .for_each(|(committee_index, validator_duties)| {
-                // Spawn a separate task for each attestation.
-                self.inner.executor.spawn_ignoring_error(
-                    self.clone().publish_attestations_and_aggregates(
-                        slot,
-                        committee_index,
-                        validator_duties,
-                        aggregate_production_instant,
-                    ),
-                    "attestation publish",
-                );
-            });
+        // Spawn a task that awaits the attestation data handle and then spawns aggregate tasks
+        let attestation_service_clone = self.clone();
+        let executor = self.inner.executor.clone();
+        self.inner.executor.spawn(
+            async move {
+                // Log an error if the handle fails and return, skipping aggregates
+                let Some(handle) = attestation_data_handle else {
+                    error!(slot = slot.as_u64(), "Failed to spawn attestation task");
+                    return;
+                };
+
+                let Ok(Some(Ok(attestation_data))) = handle.await else {
+                    error!(slot = slot.as_u64(), "Failed to spawn attestation task");
+                    return;
+                };
```

Comment:
It's a little spicy but we could combine these two statements into one like this:


```suggestion
                let Some(attestation_data) =
                    async { attestation_data_handle?.await.ok()?.and_then(Result::ok) }.await
                else {
                    error!(slot = slot.as_u64(), "Failed to spawn attestation task");
                    return;
                };
```

Reply:
This is elegant! I didn't know how to handle that before so I separate into 2 step but I know it's ugly, thanks for the comment